packages/prover/src/utils/execution.ts

```diff
@@ -0,0 +1,108 @@
+import {RLP} from "@ethereumjs/rlp";
+import {Trie} from "@ethereumjs/trie";
+import {Account} from "@ethereumjs/util";
+import {keccak256} from "ethereum-cryptography/keccak.js";
+import {Bytes32} from "@lodestar/types";
+import {ethGetBalance} from "../verified_requests/eth_getBalance.js";
+import {ELRequestPayload, ELResponse, ELProof, ELStorageProof, HexString} from "../types.js";
+import {ProofProvider} from "../proof_provider/proof_provider.js";
+import {ELRequestMethod, ELVerifiedRequestHandler} from "../interfaces.js";
+import {hexToBuffer, padLeft} from "./conversion.js";
+
+const emptyAccountSerialize = new Account().serialize();
+const storageKeyLength = 32;
+
+// eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/no-explicit-any
+const supportedELRequests: Record<string, ELVerifiedRequestHandler<any, any>> = {eth_getBalance: ethGetBalance};
+
+export async function processAndVerifyRequest({
+  payload,
+  handler,
+  proofProvider,
+}: {
+  payload: ELRequestPayload;
+  handler: ELRequestMethod;
+  proofProvider: ProofProvider;
+}): Promise<ELResponse | undefined> {
+  await proofProvider.waitToBeReady();
```

Comment:
should we race here against a timeout?

Reply:
Added an action item to the list, better to be done on multiple places in a separate PR.